### PR TITLE
fix: fix bubble list layout update on height change

### PR DESCRIPTION
**Pull request by 52cyb**
Replace the remove/insert workaround with `replace` + `dataChanged` and handle
height recalculations properly by toggling displayMarginBeginning. The old
approach of removing and inserting items forced delegate recreation but caused
unnecessary overhead and potential visual glitches. The new approach uses
QList::replace() which is more efficient, but requires explicit handling of
height updates because the view's transition logic can prevent proper re-layout when an item's height changes during a running transition.

Influence:
1. Test notification bubble replacement in normal scenarios
2. Test bubble height changes when content updates (e.g., expanding/ collapsing)
3. Verify no visual glitches during rapid notification updates
4. Test with multiple simultaneous notifications and transitions
5. Verify smooth scrolling behavior when bubble heights change
6. Test edge cases with very long notification messages

fix: 修复气泡列表高度变化时布局更新问题

用 `replace` + `dataChanged` 替换移除/插入的变通方案，并通过切换
displayMarginBeginning 正确处理高度重新计算。旧方法通过移除和插入项目
强制
委托重建，但导致不必要的开销和潜在的视觉问题。新方法使用
QList::replace()
效率更高，但需要显式处理高度更新，因为视图的过渡逻辑可能会在运行过渡期间
阻止项目高度变化时的正确重新布局。

Influence:
1. 测试正常场景下的通知气泡替换
2. 测试内容更新时气泡高度的变化（如展开/折叠）
3. 验证快速更新通知时无视觉闪烁
4. 测试多个同时发生的通知和过渡
5. 验证气泡高度变化时滚动行为流畅
6. 测试超长通知消息的边界情况

PMS: BUG-367205

## Summary by Sourcery

Improve notification bubble replacement to use in-place model updates and ensure list layout recalculates correctly when bubble heights change.

Bug Fixes:
- Fix notification bubble list failing to update layout when bubble heights change during transitions.

Enhancements:
- Use QList in-place replacement with dataChanged for bubble items instead of remove/insert to reduce overhead and avoid visual glitches during rapid updates.

### CODE REVIEW

File: panels/notification/bubble/package/BubbleDelegate.qml [hunk 1]
```diff
@@ -19,6 +19,28 @@ Item {
     property int realIndex: index < 0 ? ListView.view.count + index : index;
     
     height: bubbleContent.height
+
+    onHeightChanged: {
```
Comment (18202781743) [inline panels/notification/bubble/package/BubbleDelegate.qml]: 这个能再优化下么，减少view变化的触发，另外这个逻辑放在main.qml里，listview里，

**Review by 18202781743 — APPROVED** [review]
